src/api/app/controllers/source_controller.rb

```diff
@@ -1068,6 +1068,10 @@ def package_command_deleteuploadrev
 
   # POST /source/<project>/<package>?cmd=linktobranch
   def package_command_linktobranch
+    if @package.nil?
```

Comment:
This should rather be done wherever `@package` is set right?

Reply:
The package is set here https://github.com/openSUSE/open-build-service/blob/5396dff38018284d5be6963b9c32c1201260046a/src/api/app/controllers/source_controller.rb#L537 and there is a specific ban for `_pattern`, so...

Reply:
You will have to connect this somehow or no one after you will understand this `if`...

Reply:
I think I made it more explicit, please check it out again.